lib/hanami/view/template.rb

```diff
@@ -7,9 +7,17 @@ module View
     # @since 0.1.0
     class Template
       def initialize(template, encoding = Encoding::UTF_8)
-        # NOTE disable_escape: true is for Slim compatibility
-        # See https://github.com/hanami/assets/issues/36
-        @_template = Tilt.new(template, nil, default_encoding: encoding, disable_escape: true)
+        default_options = {default_encoding: encoding}
+
+        options = default_options
+
+        if Tilt[template] != Tilt::HamlTemplate
```

Comment:
This will trigger Tilt autoloading of the `HamlTemplate` class, which requires `haml` gem. For applications that don't bundle that gem that will cause a crash.

**With this branch checked out, from a separated directory try to do this:**

```shell
$ pwd
/Users/luca/Desktop/test

$ cat Gemfile
source "https://rubygems.org"

gem "hanami-view", path: "/Users/luca/Code/hanami/view"
```

Then start an IRB session:

```ruby
irb(main):001:0> require "hanami/view"
=> true
irb(main):002:0> Hanami::View::Template.new("index.html.erb")
Traceback (most recent call last):
       16: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
       15: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/cli.rb:27:in `dispatch'
       14: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
       13: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
       12: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
       11: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/cli.rb:424:in `exec'
       10: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/cli/exec.rb:28:in `run'
        9: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/cli/exec.rb:75:in `kernel_load'
        8: from /Users/luca/.rubies/ruby-2.5.0/lib/ruby/site_ruby/2.5.0/bundler/cli/exec.rb:75:in `load'
        7: from /Users/luca/.rubies/ruby-2.5.0/bin/irb:11:in `<top (required)>'
        6: from (irb):2
        5: from (irb):2:in `new'
        4: from /Users/luca/Code/hanami/view/lib/hanami/view/template.rb:14:in `initialize'
        3: from /Users/luca/Code/hanami/view/lib/hanami/view/template.rb:14:in `require'
        2: from /Users/luca/.gem/ruby/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:2:in `<top (required)>'
        1: from /Users/luca/.gem/ruby/2.5.0/gems/tilt-2.0.8/lib/tilt/haml.rb:2:in `require'
LoadError (cannot load such file -- haml)
```

Reply:
Right. Won't we have the same problem with `if Tilt[template] == Slim::Template`? Is slim different?

Reply:
Yes we would have the same problem, but this is a different perspective from previous comment. The first one is only about boolean logic, this one is a blocker because it would crash apps.

So `if Tilt[template] == Slim::Template` is **NOT** a viable solution.

Reply:
@jodosha what do you suggest then? Should we do something like `if File.basename(template).include? '.slim'`? Looking at the slim code I see nothing more elegant than that...

Reply:
@jodosha that would only look at the last extension, I guess that's fine in this case...